feat(test helpers): add disableFeature() test helper

Techn1x opened the pull request:
For a handful of our new features, we enable them by default in test/dev and write the majority of our tests against that. But we would still like to test certain scenarios when the feature is disabled... so I've added this as a counterpart to the `enableFeature()` test helper that is available

* added `disableFeature()` addon test helper
* added a test for toggling flags with `enableFeature()` and `disableFeature()`
* updated readme for the new test helper

Usage:
```js
test('test for my new feature, async function(assert) {
  enableFeature('feature-1');
  await visit('/');
  assert.dom('.my-new-feature').exists();

  disableFeature('feature-1');
  await settled();
  assert.dom('.my-new-feature').doesNotExist();
})
```

Techn1x commented:
@kategengler _Full disclosure_: I am participating in hacktoberfest this year so I would much appreciate it if this pull request could have the `hacktoberfest-accepted` label added to it - but I appreciate I'm towards the end of the month so if you don't get to it in time that's OK.

Also, the failing test seems to be due to unrelated observer usage in the main helper

Techn1x commented on .eslintrc.js:
```diff
@@ -16,7 +16,8 @@ module.exports = {
   rules: {
-    'ember/no-jquery': 'error'
+    'ember/no-jquery': 'error',
+    'ember/no-observers': 'warn'
```
We need this so that builds don't fail until we replace observer usage
I tried and failed here https://github.com/kategengler/ember-feature-flags/pull/98

kategengler replied:
I don't understand why you needed to change the lint rules when you don't introduce any observers here and have not updated dependencies?

Techn1x replied:
I don't quite understand either, I would have thought this lint issue would have showed up in whatever PR added the lint rule

But you can see from the failed builds in the earlier commits of this PR that it's due to this lint rule.
https://travis-ci.org/github/kategengler/ember-feature-flags/jobs/738722244

I can remove the lint warn override if you like

```
not ok 4 Chrome 62.0 - [0 ms] - ESLint | addon: addon/helpers/feature-flag.js
    ---
        actual: >
            false
        expected: >
            true
        stack: >
                at Object.<anonymous> (http://localhost:7357/assets/dummy.js:58:12)
                at runTest (http://localhost:7357/assets/test-support.js:5669:32)
                at Test.run (http://localhost:7357/assets/test-support.js:5657:8)
                at http://localhost:7357/assets/test-support.js:5874:15
                at processTaskQueue (http://localhost:7357/assets/test-support.js:5264:26)
                at http://localhost:7357/assets/test-support.js:5268:12
        message: >
            addon/helpers/feature-flag.js should pass ESLint
            
            14:5 - Don't use observers (ember/no-observers)
```

Techn1x replied:
Additionally you can see that the build failed on the 6.0 commit - opening the travis log shows it was this lint rule as well.

![Screenshot from 2020-10-26 10-13-02](https://user-images.githubusercontent.com/1049837/97122741-e2e0e700-1773-11eb-80dd-a7d22dcde6be.png)

kategengler replied:
It looks like it started failing on the cron in the scenarios that float dependencies. Instead of adding the rule here, would you mind adding it where the observers are used, using eslint-disable? Something like `/* eslint-disable ember/no-observers */` and `/* eslint-enable ember/no-observers */ wrapped around the usage of observers?

kategengler commented on addon-test-support/index.js:
```diff
@@ -1,7 +1,12 @@
   _enableFeature(owner, featureName);
 }
+
+export function disableFeature(featureName) {
```
`enableFeature` has `_enableFeature` extracted only because `_enableFeature` is used by the older `withFeature` helper -- there's no reason to do the same with `disableFeature`. I suggest splitting the two helpers into separate files (rather than toggle-feature), and implementing the whole of `disableFeature` in a `disable-feature.js` than can be reexported here `export { disableFeature } from './-private/disable-feature';`

Techn1x commented:
Thankyou for the feedback! I think I addressed your concerns about splitting the test helpers, let me know if I didn't implement it how you expected or if further changes are needed

kategengler commented:
I've added the hacktoberfest topic to the repo, so this PR will count once approved/merged.

kategengler commented:
Thank you!

sastapov-gd commented:
Hi @kategengler! Do you mind to release new minor version with this addition?
